libraries/browser-tracker-core/src/helpers/cross_domain.ts

```diff
@@ -0,0 +1,55 @@
+import { ExtendedCrossDomainLinker, ExtendedCrossDomainLinkerAttributes } from '../tracker/types';
+
+const DEFAULT_CROSS_DOMAIN_LINKER_PARAMS: ExtendedCrossDomainLinkerAttributes = {
+  sessionId: true,
+  sourceId: true,
+  sourcePlatform: false,
+  userId: false,
+  reason: false,
+};
+
+export function createCrossDomainParameterValue(
+  isExtendedFormat: boolean,
+  attributeConfiguration: ExtendedCrossDomainLinkerAttributes | undefined,
+  attributeValues: ExtendedCrossDomainLinker & {
+    /* As `reason` might be a callback, we also need to pass the event to calculate the reason value. */
+    event: Event;
+  }
+): string {
+  let crossDomainParameterValue;
+  const timestamp = new Date().getTime();
+  const config = { ...DEFAULT_CROSS_DOMAIN_LINKER_PARAMS, ...attributeConfiguration };
+  const { domainUserId, userId, sessionId, sourceId, sourcePlatform, event } = attributeValues;
+
+  const eventTarget = event.currentTarget as HTMLAnchorElement | HTMLAreaElement | null;
+  const reason = typeof config.reason === 'function' ? config.reason(event) : eventTarget?.textContent?.trim();
+
+  if (isExtendedFormat) {
+    /* Index is used by Enrich, so it should not be changed. */
+    crossDomainParameterValue = [
+      domainUserId,
+      timestamp,
+      config.sessionId && sessionId,
+      config.userId && urlSafeBase64Encode(userId || ''),
+      config.sourceId && sourceId,
+      config.sourcePlatform && sourcePlatform,
+      config.reason && urlSafeBase64Encode(reason || ''),
+    ]
+      .map((attribute) => attribute || '')
+      .join('.');
```

Comment:
Nit: this is just a visual thing but it would be nicer if the trailing dots were automatically removed at the end of the string. For instance, in case only `sessionId` is enabled in the extended config options, instead of:

```
duid.ts.sessionId....
```

the string could like like:

```
duid.ts.sessionId
```

Reply:
Done :)

Reply:
Nice!